aredis/lock.py

```diff
@@ -77,8 +78,7 @@ def __init__(self, redis, name, timeout=None, sleep=0.1,
         self.blocking = blocking
         self.blocking_timeout = blocking_timeout
         self.thread_local = bool(thread_local)
-        self.local = threading.local() if self.thread_local else dummy()
-        self.local.token = None
+        self.local = contextvars.ContextVar('token', default=None) if self.thread_local else dummy()
```

Comment:
The documentation for [contextvars](https://docs.python.org/3/library/contextvars.html) has the following disclaimer:
>Important: Context Variables should be created at the top module level and never in closures. Context objects hold strong references to context variables which prevents context variables from being properly garbage collected.

Declaring the context variable in the `__init__` method may result in a memory leak if my understanding is correct. I propose to declare it outside of the class definition. For exampe:
```python
token_var = contextvars.ContextVar('token', default=None)


class Lock:
  def __init__(...):
  self.local = token_var if self.thread_local else dummy()
  ...
```

Reply:
On second thought, moving the contextvar definition to the module level will cause tokens to bleed between lock instances created by the same co-routine. That's a bad idea. Sorry, you can ignore my proposal.

Reply:
> On second thought, moving the contextvar definition to the module level will cause tokens to bleed between lock instances created by the same co-routine. That's a bad idea. Sorry, you can ignore my proposal.

Sorry, github didn't send me a msg and i saw your code review so late.